src/storage/txn/scheduler.rs

```diff
@@ -404,6 +424,32 @@ impl<E: Engine> Scheduler<E> {
         self.release_lock(&tctx.lock, cid);
     }
 
+    /// Event handler for the request of waiting for lock
+    fn on_wait_for_lock(
+        &mut self,
+        cid: u64,
+        start_ts: u64,
+        pr: ProcessResult,
+        lock: lock_manager::Lock,
+        is_first_lock: bool,
+    ) {
+        debug!("command waits for lock released"; "cid" => cid);
+        let tctx = self.dequeue_task_context(cid);
+        SCHED_STAGE_COUNTER_VEC
+            .with_label_values(&[tctx.tag, "lock_wait"])
+            .inc();
+        // TODO: timeout config
+        self.waiter_mgr_scheduler.as_ref().unwrap().wait_for(
```

Comment:
Is it safe to unwrap?

Reply:
Yes. If `pessimitic-txn` is not enabled,`acquire_pessimistic_lock` is never called, and the `WaitForLock` message is never sent to the scheduler.
If it is enabled, `waiter_mgr_scheduler` must be some.